modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/dao/JDBCResourceDAO.java

```diff
@@ -376,4 +381,155 @@ public boolean updateResource(String applicationKey, String resourceType, String
+            addContainerPreparedStatement = connection.prepareStatement(SQLConstants.ADD_CONTAINERS);
+            addProxiesPreparedStatement = connection.prepareStatement(SQLConstants.ADD_SERVICE_PROXIES);
+
+            addDeploymentPreparedStatement.setString(1, deploymentConfig.getDeploymentName());
+            addDeploymentPreparedStatement.setInt(2, deploymentConfig.getReplicas());
+            addDeploymentPreparedStatement.executeUpdate();
+            connection.commit();
+
+            List<Container> containerList = deploymentConfig.getContainers();
+            for (Container container : containerList) {
+                addContainerPreparedStatement.setString(1, container.getBaseImageName());
+                addContainerPreparedStatement.setString(2, container.getBaseImageVersion());
+                addContainerPreparedStatement.setString(3, deploymentConfig.getDeploymentName());
+                addContainerPreparedStatement.executeUpdate();
+                connection.commit();
+
+                List<ServiceProxy> serviceProxyList = container.getServiceProxies();
+
+                for (ServiceProxy serviceProxy : serviceProxyList) {
+                    addProxiesPreparedStatement.setString(1, serviceProxy.getServiceName());
+                    addProxiesPreparedStatement.setString(2, serviceProxy.getServiceProtocol());
+                    addProxiesPreparedStatement.setInt(3, serviceProxy.getServicePort());
+                    addProxiesPreparedStatement.setInt(4, serviceProxy.getServiceBackendPort());
+                    addProxiesPreparedStatement.setString(5, container.getBaseImageName());
+                    addProxiesPreparedStatement.setString(6, container.getBaseImageVersion());
+                    addProxiesPreparedStatement.setString(7, deploymentConfig.getDeploymentName());
+                    addProxiesPreparedStatement.executeUpdate();
+                    connection.commit();
+                }
+            }
+
+        } catch (SQLException e) {
+            String message = "Error while adding database for deployement config : "
+                    + deploymentConfig.getDeploymentName();
+            log.error(message, e);
+            throw new AppFactoryException(message, e);
+        } finally {
+            AppFactoryDBUtil.closePreparedStatement(addDeploymentPreparedStatement);
+            AppFactoryDBUtil.closePreparedStatement(addContainerPreparedStatement);
+            AppFactoryDBUtil.closePreparedStatement(addProxiesPreparedStatement);
+            AppFactoryDBUtil.closeConnection(connection);
+        }
+    }
+
+    /**
+     * Get deployement config
+     *
+     * @param applicationId id of the application
+     * @param stage         stage of the application
+     * @return deployement config
+     */
+    public DeploymentConfig getDeploymentConfig(String applicationId, String stage) throws AppFactoryException {
```

Comment:
This method is too long.. Break into small methods such that it will improve the readability and the manageability.